### PR TITLE
Change PowerDistributor to ignore components with NaN metrics

### PR DESCRIPTION
**Pull request by ela-kotulska-frequenz**
Each float metric returned from api can be float("NaN"). If that happen PowerDistributor would fail because it is impossible to make math calcuation on NaN value.
This PR is just a quick fix of this problem.
* if power bounds are NaN, then it tries to replace them with corresponding power bounds from adjacent component. If these components are also NaN, then it ignores battery.
* if other metrics metrics are NaN then it ignores battery.

More sophisticated solution should be done in the future, maybe use resampler.

### CODE REVIEW

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (leandro-lucarella-frequenz) [inline src/frequenz/sdk/actor/power_distributing/power_distributing.py:600]: Not sure if it's worth it, but if it should never happen maybe log an error in case there is some problem that can be detected earlier.

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (leandro-lucarella-frequenz) [inline src/frequenz/sdk/actor/power_distributing/power_distributing.py]: Not sure if this should be a warning, couldn't it spam the logs too much if some device is sending `NaN`s for a while? Isn't this sort of *normal* behavior?

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (leandro-lucarella-frequenz) [inline src/frequenz/sdk/actor/power_distributing/power_distributing.py]: Just for fun :D, this could also be written:
```py
         if not any(map(isnan, replaceable_metrics)):
```

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (leandro-lucarella-frequenz) [inline src/frequenz/sdk/actor/power_distributing/power_distributing.py:572]: Maybe it would be nice to have a comment on this method even if it's private to mention that this is dealing mainly with `NaN`s and how they are being handled?

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (leandro-lucarella-frequenz) [inline src/frequenz/sdk/actor/power_distributing/power_distributing.py]: Same comment about confirming if we really want this to be a warning.

**Review by leandro-lucarella-frequenz — DISMISSED** [review]
I didn't look into the tests in depth, and other than the small comments, LGTM. They are all minor, so I leave it to you if you want to change any of them, so giving an approval.

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/actor/power_distributing/power_distributing.py:600]: This moment when you get `Not sure if it's worth it` comment and you realize it is bug :D
https://github.com/frequenz-floss/frequenz-sdk-python/pull/247/commits/198f05f71a3607cfa496133813c7f79caebaf600

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/actor/power_distributing/power_distributing.py]: Done, You are right. I changed it to debug log

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/actor/power_distributing/power_distributing.py]: Haha Thanks, done :D

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/actor/power_distributing/power_distributing.py:572]: Described

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/actor/power_distributing/power_distributing.py]: We don't. Changed to debug.

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (leandro-lucarella-frequenz) [inline src/frequenz/sdk/actor/power_distributing/power_distributing.py]: Not important as this is private and it won't get rendered, but just a reminder for the cases where it will:

```suggestion
        We can't do math operations on "NaN".        
        So check all the metrics and:
        * if power bounds are NaN, then try to replace it with the corresponding
          power bounds from the adjacent component. If metric in the adjacent component is
          also NaN, then return None.
        * if other metrics are NaN then return None. We can't assume anything for other
          metrics.
```

**Review by leandro-lucarella-frequenz — DISMISSED** [review]

**Comment by ela-kotulska-frequenz** [comment]
Rebased and changed formatting.

File: src/frequenz/sdk/actor/power_distributing/power_distributing.py
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/actor/power_distributing/power_distributing.py]: Done

**Comment by ela-kotulska-frequenz** [comment]
> I didn't look into the tests in depth

Tests are copy-paste. :( 
I am not proud of them but it was changed so many times... I hope they would be improved when we do PowerManger.

**Pull request merged**
